Backpacks.cs

```diff
@@ -569,19 +572,28 @@ public void Initialize()
                 else
                 {
                     // Force-close since we are re-initializing
-                    ForceClose();
+                    ForceCloseAllLooters();
                 }
 
                 var ownerPlayer = FindOwnerPlayer()?.Object as BasePlayer;
-
-                _itemContainer.capacity = GetCapacity();
                 _itemContainer.entityOwner = ownerPlayer;
 
+                var allowedCapacity = GetAllowedCapacity();
                 if (!_initialized)
                 {
                     _itemContainer.isServer = true;
                     _itemContainer.allowedContents = ItemContainer.ContentsType.Generic;
                     _itemContainer.GiveUID();
+                    _itemContainer.capacity = allowedCapacity;
+
+                    if (_itemDataCollection.Count > allowedCapacity)
```

Comment:
One possibility which doesn't seem covered right now is that they don't have many items, but for some reason they're stored in the last few slots which wouldn't be available anymore.

So instead of checking the count of items, I'd maybe check if there is any item which is supposed to be in a slot which is no longer available.
Probably best to write a method to check that, for separation & reusability.

Reply:
That was a case I continuously tested while developing. Testing again confirms it works. For example, when the capacity is 12 (2 rows of 6), I put in 5 items, some on the first row, some one the second. Then I change permission to reduce the capacity to 6 and re-open the backpack. The items on the second row are automatically moved to the first row in the same order.

I haven't worked on this for several weeks so I couldn't remember why this works, and it's not obvious from looking at the code. After some testing, it turns out this is handled by the following branch.

```
else if (_itemContainer.capacity > allowedCapacity)
{
    // The capacity will be reduced later when the backpack is opened by the owner
    hasOverflowingItems = true;
}
```

This actually causes the `hasOverflowingItems` flag to be set every time the capacity is reduced by a permission change, so the `HandleOverflowingItems()` method will be called on the next `Open()` from the owner even if there is no capacity issue. This is probably fine and could be clarified with some renames. However, I thought it through and tested it some more, and I found one issue, which is that if an admin views the backpack, the non-overflowing items that are beyond the capacity may be hidden. I will try to fix this in the next revision.